.gitignore

```diff
@@ -50,3 +50,6 @@ qtcreator-*
 # Catkin custom files
 CATKIN_IGNORE
 
+# uwrt_mars_rover_hw_bridge shouldnt be added as a git submodule
+uwrt_mars_rover_hw_bridge_pkg/
```

Comment:
you should ignore the git submodule itself, not the entire ros pkg. in this case uwrt_mars_rover_hw_bridge_pkg/uwrt_mars_rover_hw_bridge


the ros pkg should have everything other ros packages have. ie. package.xml, cmakelists, etc.